### PR TITLE
Add instructions for enabling network booting for a Pi 3B+ and 4.

### PR DESCRIPTION
**Pull request by Malvineous**
The net boot instructions are missing the required steps to enable network boot on a Raspberry Pi 3B+ and 4.  This patch adds these instructions.

### CODE REVIEW

File: hardware/raspberrypi/bootmodes/net_tutorial.md [hunk 1]
```diff
@@ -35,6 +35,18 @@ Ensure the output `0x3020000a` is correct.
 
 The client configuration is almost done. The final thing to do is to remove the `program_usb_boot_mode` line from `config.txt` (make sure there is no blank line at the end). You can do this with `sudo nano /boot/config.txt`, for example. Finally, shut the client Raspberry Pi down with `sudo poweroff`.
 
+## Client configuration (Raspberry Pi 3B+ / 4)
+
+These instructions only apply to the Raspberry Pi 3B+ and 4.
```
Comment (lurch) [inline hardware/raspberrypi/bootmodes/net_tutorial.md:40]: The Raspberry Pi 3B+ doesn't have a bootloader eeprom...

File: hardware/raspberrypi/bootmodes/net_tutorial.md [hunk 2]
```diff
@@ -35,6 +35,18 @@ Ensure the output `0x3020000a` is correct.
...
+
+Follow the instructions on the [Pi 4 Bootloader Configuration](../bcm2711_bootloader_config.md) page to alter the boot order to include network booting.  In short:
+
+```bash
+sudo -E rpi-eeprom-config --edit
+```
+
+Set `BOOT_ORDER` to `0xf21` or similar, save, and reboot.  The updated EEPROM settings will be saved onto the SD card and programmed to the EEPROM during the next boot.  After this reboot the settings will become permanent and the SD card is no longer required.
```
Comment (lurch) [inline hardware/raspberrypi/bootmodes/net_tutorial.md:48]: I think there might be a `raspi-config` option to do this now, which is easier than using `rpi-eeprom-config` directly?

Ah yes: https://www.raspberrypi.org/documentation/configuration/raspi-config.md

**Comment by andrum99** [comment]
We probably also want to remove `Before you do anything else, make sure you have run `sudo raspi-config` and expanded the root file system to take up the entire SD card.` under `Server configuration` - Raspberry Pi OS has automatically expanded the root filesystem since before netboot was a thing on the Pi 🤨

File: hardware/raspberrypi/bootmodes/net_tutorial.md (hunk 1 above)
Comment (Malvineous) [reply hardware/raspberrypi/bootmodes/net_tutorial.md:40]: Oh you're quite right, how silly of me!  I will see if I can find out what is required on the 3B+ and update this PR.

Comment (lurch) [reply hardware/raspberrypi/bootmodes/net_tutorial.md:40]: IIRC the 3B+ supports network (and USB) booting out-of-the-box?

Comment (andrum99) [reply hardware/raspberrypi/bootmodes/net_tutorial.md:40]: > IIRC the 3B+ supports network (and USB) booting out-of-the-box?

Correct. There's no configuration needed - if it hasn't found an SD card in 5 seconds it moves to USB host boot and tries USB MSD and net, as already stated on this page.

**Comment by aallan** [comment]
I'm probably going to be freezing the current markdown-based documentation tree, and deleting any pending PRs, at the end of the week (see #1911 for more information as to why). So if we're going to merge this, probably need to finalise it before Wednesday or Thursday.

**Comment by aallan** [comment]
I'm going to be freezing the current Markdown-based repo for PRs on Monday morning, see #1911 for details of what's going on. If this PR isn't merged before then it will unfortunately be deleted. Although we'll be happy to get a new PR against the new Asciidoc-based repo after that goes public — which will probably at the start of August.

**Comment by aallan** [comment]
There is now an alternative PR, either this PR or #1942 should be merged.

**Comment by aallan** [comment]
#1942 was merged instead of this PR.

**Pull request closed without merging**